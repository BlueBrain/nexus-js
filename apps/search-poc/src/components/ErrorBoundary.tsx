
import React from 'react';


class ErrorBoundary extends React.Component<{}, {
  hasError: boolean;
}> {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  componentDidCatch() {
    this.setState({ hasError: true });
  }

  render() {
    if (this.state.hasError) {
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children;
  }
}


export default ErrorBoundary;
