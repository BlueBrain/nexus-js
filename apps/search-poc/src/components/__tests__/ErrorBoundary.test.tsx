import * as renderer from "react-test-renderer";
import ErrorBoundary from '../ErrorBoundary';
import React from "react";
import { mount } from "enzyme";

const MockComponent = () => null;
it('should display an ErrorMessage if wrapped component throws', () => {
    const wrapper = mount(
      <ErrorBoundary>
        <MockComponent />
      </ErrorBoundary>
    );

    const error = new Error('test');

    wrapper.find(MockComponent).simulateError(error);

    expect(wrapper).toMatchSnapshot();
});