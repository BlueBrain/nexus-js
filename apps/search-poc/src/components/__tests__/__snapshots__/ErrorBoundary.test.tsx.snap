// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display an ErrorMessage if wrapped component throws 1`] = `
<ErrorBoundary>
  <h1>
    Something went wrong.
  </h1>
</ErrorBoundary>
`;
